packages/stream-tile-handler/src/tile-document-handler.ts

```diff
@@ -70,7 +70,7 @@ export class TileDocumentHandler implements StreamHandler<TileDocument> {
         const isSigned = StreamUtils.isSignedCommit(commit)
         if (isSigned) {
             payload = (await context.ipfs.dag.get(commit.link, { timeout: IPFS_GET_TIMEOUT })).value
-            await this._verifySignature(commit, context, payload.header.controllers[0])
+            await this._verifySignature(commit, {cid: cid}, context, payload.header.controllers[0])
```

Comment:
Wouldn't this mean that a signed genesis will fail if you rotate keys? i.e. we need timestamp for genesis commit as well.

Reply:
Right. Just passed time information there as well.

There is a one more issue with that though. For a genesis with no anchor, there is no way to deduce its timestamp. It could be created any time `(-∞; now]`. So, if we always lean towards `now`, there is a possibility of accidental signature invalidation for non-anchored genesis. Probably, one could enforce creating an anchor for a signed genesis.

Reply:
Seems like this is exactly the same as non-anchored signed commits in general.

Reply:
I was going to disagree, telling that the interval now is `(prevAnchorTimestamp; now]`, but it does not make any difference. The same, agree.